esiclient/v1/node_network.py

```diff
@@ -264,21 +264,40 @@ def take_action(self, parsed_args):
         ironic_client = self.app.client_manager.baremetal
         neutron_client = self.app.client_manager.network
 
-        node = None
-        port = None
+        current_node = ironic_client.node.get(node_uuid)
 
-        with concurrent.futures.ThreadPoolExecutor() as executor:
-            f1 = executor.submit(ironic_client.node.get, node_uuid)
-            f2 = executor.submit(neutron_client.find_port, port_uuid)
-            node = f1.result()
-            port = f2.result()
-
-        if not port:
+        if current_node is None:
             raise exceptions.CommandError(
-                "ERROR: Port {1} not attached to node {0}".format(
-                    node.name, port_uuid))
+                "ERROR: The node {0} is not available".format(
+                    node_uuid))
+
+        if port_uuid:
+            port = neutron_client.find_port(port_uuid)
+            if not port:
+                raise exceptions.CommandError(
+                    "ERROR: Port {1} not attached to node {0}".format(
+                        current_node.name, port_uuid))
+        else:
+            ports = ironic_client.port.list(node=node_uuid, detail=True)
+
+            mapped_node_port_list = []
+            for port in ports:
+                if port.internal_info.get("tenant_vif_port_id"):
+                    mapped_node_port_list.append(port)
+            if(len(mapped_node_port_list) == 0):
+                raise exceptions.CommandError(
+                    "ERROR: Node {0} is not assosiated with any port".format(
+                        current_node.name))
+            elif(len(mapped_node_port_list) > 1):
+                raise exceptions.CommandError(
+                    "ERROR: Multiple ports for {0} Specify the port".format(
```

Comment:
For parallelism in error messages, I'd make this `ERROR: Node {0} is associated with multiple ports. Port must be specified with --port`